### DISCUSSION TITLE
Switching back to npm 6 from npm 7 results in install error

### DISCUSSION THREAD

**Discussion by s2danii** [npm/feedback#347] [CLI] [answered]
I believe npm 7 is meant to be fully backwards compatible with npm 6 but I'm seeing some slight changes in my package-lock.json when I switch between the two that is causing an install error.

This is an example I'm seeing in package-lock.json under `dependencies` for this library:

**npm 6 correct format:**
```
    "webpack-raphael": {
      "version": "2.1.4",
      "resolved": "https://registry.npmjs.org/webpack-raphael/-/webpack-raphael-2.1.4.tgz",
      "integrity": "sha1-urmFNtYox2zSQcuj0KbMdkszBD4=",
      "requires": {
        "eve": "git://github.com/adobe-webplatform/eve.git#eef80ed"
      }
    },
```

**npm 7 format:**
```
    "node_modules/webpack-raphael": {
      "version": "2.1.4",
      "resolved": "https://registry.npmjs.org/webpack-raphael/-/webpack-raphael-2.1.4.tgz",
      "integrity": "sha1-urmFNtYox2zSQcuj0KbMdkszBD4=",
      "dependencies": {
        "eve": "git://github.com/adobe-webplatform/eve.git#eef80ed"
      }
    },
```

**installing using npm 7** format package-lock, **but in npm 6** (I'm on 6.14.0) will generate a new package-lock in npm 6 format, but changes the format of the git source for certain dependencies. i.e. `"git://github.com/adobe-webplatform/eve.git#eef80ed"` becomes  `"eve@git://github.com/adobe-webplatform/eve.git#eef80ed"`:

```
   "webpack-raphael": {
      "version": "2.1.4",
      "resolved": "https://registry.npmjs.org/webpack-raphael/-/webpack-raphael-2.1.4.tgz",
      "integrity": "sha1-urmFNtYox2zSQcuj0KbMdkszBD4=",
      "requires": {
        "eve": "eve@git://github.com/adobe-webplatform/eve.git#eef80ed"
      }
    },
```

now if I delete my node_modules and try installing in npm 6 from this new package-lock, I get the following error:
```
npm ERR! code ENOLOCAL
npm ERR! Could not install from "node_modules/webpack-raphael/eve@git:/github.com/adobe-webplatform/eve.git#eef80ed" as it does not contain a package.json file.
```

**Comment by MylesBorins** [comment, marked as answer]
hey @s2danii thanks for opening this discussion. This appears to be a bug report for the npm CLI, so the next step for you would be to open an issue over at https://github.com/npm/cli
